mitodl/mitxpro
# Added retry for timed-out Mailgun API requests

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1460 

#### What's this PR do?
Adds a retry for Mailgun API requests that time out

#### How should this be manually tested?
I tested with the the CI Mailgun credentials in my settings. This is the code I used to test the relevant API function directly:

```python
from mitxpro.utils import now_in_utc
from datetime import timedelta
from sheets.mail_api import get_bulk_assignment_messages
#
now = now_in_utc()
past = now - timedelta(days=7)
list(get_bulk_assignment_messages(begin=past, end=now))
```

You can't force the Mailgun API to have a timeout, but in my case there happened to be a timeout by happy coincidence, and the code handled it gracefully (see screenshot below). If you want to test the timeout yourself, you can change the URL to use httpbin and force a 504 response or something like that

#### Screenshots (if appropriate)
<img width="764" alt="ss 2020-01-09 at 12 18 27 " src="https://user-images.githubusercontent.com/14932219/72089809-b938fb00-32da-11ea-9891-92646e59326d.png">
